lib/elixir_make/artefact.ex

```diff
@@ -187,11 +213,25 @@ defmodule ElixirMake.Artefact do
   @doc """
   Returns the url for the current target.
   """
-  def current_target_url(config, precompiler, nif_version) do
+  def current_target_url(config, precompiler, current_nif_version) do
     case precompiler.current_target() do
       {:ok, current_target} ->
+        nif_versions =
+          config[:make_precompiler_nif_versions] ||
+            [versions: []]
+
+        versions = nif_versions[:versions]
+
+        nif_version_to_use =
+          if current_nif_version in versions do
+            current_nif_version
+          else
+            fallback_version = nif_versions[:fallback_version] || (&fallback_version/3)
+            fallback_version.(current_target, current_nif_version, versions)
+          end
```

Comment:
@cocoa-xu I just moved the conditional outside. I think that, if the `current_nif_version` matches, we should never invoke the fallback. Makes sense right? If you are happy, feel free to merge it, otherwise I do it when I am back home!

Reply:
Oh, I thought this was a elixir-nx project, it isn't. :) I will go ahead and merge it.

Reply:
> @cocoa-xu I just moved the conditional outside. I think that, if the `current_nif_version` matches, we should never invoke the fallback. Makes sense right? If you are happy, feel free to merge it, otherwise I do it when I am back home!

Ah you're right!

> Oh, I thought this was a elixir-nx project, it isn't. :) I will go ahead and merge it.

:D